src/data/data.cc

```diff
@@ -350,30 +350,44 @@ void MetaInfo::Validate() const {
         << "Invalid group structure.  Number of rows obtained from groups "
            "doesn't equal to actual number of rows given by data.";
   }
+  auto check_device = [device](HostDeviceVector<float> const &v) {
```

Comment:
Should this be a proper function?

Reply:
@hcho3 I don't have a preference.  Maybe a member function of `HostDeviceVector`?  Or a general function in `device_helpers.cuh`?

Reply:
It can go in `device_helpers.cuh`.

Reply:
@hcho3 No, my bad.  data.cc can not include a `cuh` file.

Reply:
Okay, I suppose we can merge this PR as it is.